models/PyCryptoBot.py

```diff
@@ -861,108 +858,65 @@ def setNoSellAtLoss(self, flag):
 
             if self.sellLowerPcnt() != None:
-                txt = '      Sell Lower Pcnt : ' + str(
-                    not self.disableFailsafeLowerPcnt()) + '  --disablefailsafelowerpcnt'
-                Logger.info('|  ' + txt + (' ' * (75 - len(txt))) + ' | ')
+                textBox.line('Sell Lower Pcnt', str(not self.disableFailsafeLowerPcnt()) + '  --disablefailsafelowerpcnt')
 
             if self.sellUpperPcnt() != None:
-                txt = '      Sell Upper Pcnt : ' + str(
-                    not self.disableFailsafeLowerPcnt()) + '  --disableprofitbankupperpcnt'
-                Logger.info('|  ' + txt + (' ' * (75 - len(txt))) + ' | ')
-
-            txt = ' Candlestick Reversal : ' + str(
-                not self.disableProfitbankReversal()) + '  --disableprofitbankreversal'
-            Logger.info('|  ' + txt + (' ' * (75 - len(txt))) + ' | ')
-
-            txt = '             Telegram : ' + str(not self.disabletelegram) + '  --disabletelegram'
-            Logger.info('|  ' + txt + (' ' * (75 - len(txt))) + ' | ')
-
-            txt = '                  Log : ' + str(not self.disableLog()) + '  --disablelog'
-            Logger.info('|  ' + txt + (' ' * (75 - len(txt))) + ' | ')
-
-            txt = '              Tracker : ' + str(not self.disableTracker()) + '  --disabletracker'
-            Logger.info('|  ' + txt + (' ' * (75 - len(txt))) + ' | ')
+                textBox.line('Sell Upper Pcnt', str(not self.disableFailsafeLowerPcnt()) + '  --disableprofitbankupperpcnt')
 
-            txt = '     Auto restart Bot : ' + str(self.autoRestart()) + '  --autorestart'
-            Logger.info('|  ' + txt + (' ' * (75 - len(txt))) + ' | ')
+            textBox.line('Candlestick Reversal', str(not self.disableProfitbankReversal()) + '  --disableprofitbankreversal')
+            textBox.line('Telegram', str(not self.disabletelegram) + '  --disabletelegram')
+            textBox.line('Log', str(not self.disableLog()) + '  --disablelog')
+            textBox.line('Tracker', str(not self.disableTracker()) + '  --disabletracker')
+            textBox.line('Auto restart Bot', str(self.autoRestart()) + '  --autorestart')
 
             if self.getBuyMaxSize():
-                txt = '         Max Buy Size : ' + str(self.getBuyMaxSize()) + '  --buymaxsize <size>'
-                Logger.info('|  ' + txt + (' ' * (75 - len(txt))) + ' | ')
+                textBox.line('Max Buy Size', str(self.getBuyMaxSize()) + '  --buymaxsize <size>')
 
-            if self.disablebuyema and self.disablebuymacd :
-                Logger.info('| WARNING : EMA and MACD indicators disabled, no buy events will happen        |')
+            if self.disablebuyema and self.disablebuymacd:
+                textBox.center('WARNING : EMA and MACD indicators disabled, no buy events will happen')
 
-            Logger.info('================================================================================')
+            textBox.doubleLine()
 
```

Comment:
actually, maybe the box generation should go into the TextBoxHelper file as a method to clean this file up? It would make it easier to read the main file. WDYT?

Reply:
like:

```
if banner:
  textbox.generate_banner()
```

Reply:
I would definitely make it easier to read, but I wasn't able to get it to work since I'm new to Python. When I tried from within the class, I wasn't able to use the methods from PyCryptoBot.py, and importing it caused a circular import. Probably need to separate out the methods to a separate file.
Any thoughts?

Reply:
Create a new method in the PyCryptoBot class, like:

```
    def _generate_banner(self) -> None:
        textBox = TextBox(80, 26)
        textBox.singleLine()
        textBox.center('Python Crypto Bot')
        textBox.singleLine()
        textBox.line('Release', self.getVersionFromREADME())
        textBox.singleLine()

        if self.isVerbose():
            textBox.line('Market', self.getMarket())
            textBox.line('Granularity', str(self.getGranularity()) + ' seconds')
            textBox.singleLine()

        if self.isLive():
            textBox.line('Bot Mode', 'LIVE - live trades using your funds!')
        else:
            textBox.line('Bot Mode', 'TEST - test trades using dummy funds :)')

        textBox.line('Bot Started', str(datetime.now()))
        textBox.doubleLine()

        if self.sellUpperPcnt() != None:
            textBox.line('Sell Upper', str(self.sellUpperPcnt()) + '%')

        if self.sellLowerPcnt() != None:
            textBox.line('Sell Lower', str(self.sellLowerPcnt()) + '%')

        if self.trailingStopLoss() != None:
            textBox.line('Trailing Stop Loss', str(self.trailingStopLoss()) + '%')

        textBox.line('Sell At Loss', str(self.allowSellAtLoss()) + '  --sellatloss ' + str(self.allowSellAtLoss()))
        textBox.line('Sell At Resistance', str(self.sellAtResistance()) + '  --sellatresistance')
        textBox.line('Trade Bull Only', str(not self.disableBullOnly()) + '  --disablebullonly')
        textBox.line('Buy Near High', str(not self.disableBuyNearHigh()) + '  --disablebuynearhigh')
        textBox.line('Use Buy MACD', str(not self.disableBuyMACD()) + '  --disablebuymacd')
        textBox.line('Use Buy EMA', str(not self.disableBuyEMA()) + '  --disablebuyema')
        textBox.line('Use Buy OBV', str(not self.disableBuyOBV()) + '  --disablebuyobv')
        textBox.line('Use Buy Elder-Ray', str(not self.disableBuyElderRay()) + '  --disablebuyelderray')
        textBox.line('Sell Fibonacci Low', str(not self.disableFailsafeFibonacciLow()) + '  --disablefailsafefibonaccilow')

        if self.sellLowerPcnt() != None:
            textBox.line('Sell Lower Pcnt', str(not self.disableFailsafeLowerPcnt()) + '  --disablefailsafelowerpcnt')

        if self.sellUpperPcnt() != None:
            textBox.line('Sell Upper Pcnt', str(not self.disableFailsafeLowerPcnt()) + '  --disableprofitbankupperpcnt')

        textBox.line('Candlestick Reversal', str(not self.disableProfitbankReversal()) + '  --disableprofitbankreversal')
        textBox.line('Telegram', str(not self.disabletelegram) + '  --disabletelegram')
        textBox.line('Log', str(not self.disableLog()) + '  --disablelog')
        textBox.line('Tracker', str(not self.disableTracker()) + '  --disabletracker')
        textBox.line('Auto restart Bot', str(self.autoRestart()) + '  --autorestart')

        if self.getBuyMaxSize():
            textBox.line('Max Buy Size', str(self.getBuyMaxSize()) + '  --buymaxsize <size>')

        if self.disablebuyema and self.disablebuymacd:
            textBox.center('WARNING : EMA and MACD indicators disabled, no buy events will happen')

        textBox.doubleLine()
```

And change lines 860ff to:
```    def startApp(self, account, last_action='', banner=True):
        if banner:
            self._generate_banner()
        # run the first job immediately after starting
        if self.isSimulation():
```

Reply:
Oh and figure out what to do with lines 959-967 :)

Reply:
For lines 959-967, would it be wasteful to create another instance of the TextBox class, just for that?

Reply:
I'd rather create another instance and have it more readable than the other way round. it's not like this is an object that uses gigs of memory.

Reply:
Fair enough. I'll make these changes.

Reply:
Wow. That really made a difference in the readability...

Reply:
Agree. Much better now. Thank you for your effort!